htlcswitch/interfaces.go

```diff
@@ -357,13 +357,17 @@ type InterceptedPacket struct {
 	// IncomingAmount is the amount of the accepted htlc.
 	IncomingAmount lnwire.MilliSatoshi
 
-	// CustomRecords are user-defined records in the custom type range that
-	// were included in the payload.
-	CustomRecords record.CustomSet
+	// InOnionCustomRecords are user-defined records in the custom type
+	// range that were included in the payload.
+	InOnionCustomRecords record.CustomSet
```

Comment:
Hmm, the stuff in `record` is actually "official", in that we're setting custom records, but these are protocols that are fully defined. While `lnwire.CustomRecords` is in place so we can properly merge official records with custom records, and not have them clobber each other when we go to encode/decode.

Reply:
Hmm, actually, I just noticed that these should all be `lnwire.CustomRecords`... That's the type we'll actually want to use, since we added all those additional validation methods to them.

Reply:
Case in point :). What's the difference? Does one of those types refine the other one (is it a subset/more restrictive)?